extensions/pkg/controller/worker/machines_test.go

```diff
@@ -94,187 +104,149 @@ var _ = Describe("Machines", func() {
+				Expect(actual).NotTo(Equal(hash))
 			})
 
 			It("when changing machine type", func() {
 				p.MachineType = "small"
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing machine image name", func() {
 				p.MachineImage.Name = "new-image"
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing machine image version", func() {
 				p.MachineImage.Version = "new-version"
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing volume type", func() {
 				t := "xl"
 				p.Volume.Type = &t
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing volume size", func() {
 				p.Volume.Size = "100Mi"
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing provider config", func() {
 				p.ProviderConfig.Raw = nil
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing the kubernetes major/minor version of the worker pool version", func() {
 				p.KubernetesVersion = pointer.String("1.3.3")
-				v, err = WorkerPoolHash(*p, cluster)
 			})
 
 			It("when changing the kubernetes major/minor version of the control plane version", func() {
-				v, err = WorkerPoolHash(*p, &extensionscontroller.Cluster{
-					Shoot: &gardencorev1beta1.Shoot{
-						Spec: gardencorev1beta1.ShootSpec{
-							Kubernetes: gardencorev1beta1.Kubernetes{
-								Version: "1.3.3",
-							},
-						},
-					},
-				})
+				c.Shoot.Spec.Kubernetes.Version = "1.3.3"
 			})
 
-			It("when adding additionalData", func() {
-				v, err = WorkerPoolHash(*p, cluster, "some-additional-data")
+			It("when changing the CRI configurations", func() {
+				c.Shoot.Spec.Provider = gardencorev1beta1.Provider{Workers: []gardencorev1beta1.Worker{
+					{Name: "test-worker", CRI: &gardencorev1beta1.CRI{Name: gardencorev1beta1.CRINameContainerD}}}}
 			})
 
-			It("when changing the CRI configurations", func() {
-				v, err = WorkerPoolHash(*p, &extensionscontroller.Cluster{
-					Shoot: &gardencorev1beta1.Shoot{
-						Spec: gardencorev1beta1.ShootSpec{
-							Kubernetes: gardencorev1beta1.Kubernetes{
-								Version: "1.2.4",
-							},
-							Provider: gardencorev1beta1.Provider{Workers: []gardencorev1beta1.Worker{
-								{Name: "test-worker", CRI: &gardencorev1beta1.CRI{Name: gardencorev1beta1.CRINameContainerD}}}},
-						},
-					},
-				})
+			It("when a shoot CA rotation is triggered", func() {
+				newRotationTime := metav1.Time{Time: lastRotationInitiation.Add(time.Hour)}
+				c.Shoot.Status.Credentials.Rotation.CertificateAuthorities.LastInitiationTime = &newRotationTime
+			})
+
+			It("when a shoot CA rotation is triggered for the first time (lastInitiationTime was nil)", func() {
+				var err error
+				credentialStatusWithInitiatedRotation := c.Shoot.Status.Credentials.DeepCopy()
+				c.Shoot.Status.Credentials = nil
+				hash, err = WorkerPoolHash(p, c)
+				Expect(err).ToNot(HaveOccurred())
+
+				c.Shoot.Status.Credentials = credentialStatusWithInitiatedRotation
```

Comment:
What is the purpose of this test when we already have 

```
			It("when a shoot CA rotation is triggered", func() {
```

?

Reply:
I wrote tests first and I wanted those two cases to be covered:
- there was already a rotation
- there has not yet been a rotation
That's why I wrote them.

We can delete one if you think it's confusing.

Reply:
OK, thanks for the explanation. It makes sense when knowing the development flow, but from reading afterwards (or even months later) it is questionable/confusing 😄 So I would indeed vote for dropping it unless it provides additional value compared to the other already existing test

Reply:
The way the tests are written right now it would for example catch if those nil checks were missing:
```
if status := cluster.Shoot.Status; status.Credentials != nil && status.Credentials.Rotation != nil &&
		status.Credentials.Rotation.CertificateAuthorities != nil &&
		status.Credentials.Rotation.CertificateAuthorities.LastInitiationTime != nil {
```

Reply:
Then make it more clear that the purpose of this case is to verify the `nil` checks

Reply:
I don't want to argue over this nit but actually I do not know what to rename it to. 😄 
It is named `lastInitiationTime was nil` which is exactly the test I would expect to fail if we had missing nil checks. Also it is named after which behaviour it tests which makes sense I guess.

Maybe you want me to delete the other test and only test that it works when going from nil -> any value? That's how we did it for the the cri case?

Reply:
OK, let's simply keep it as is for now.